src/shell.js

```diff
@@ -43,6 +43,20 @@ var Module = typeof {{{ EXPORT_NAME }}} !== 'undefined' ? {{{ EXPORT_NAME }}} :
 #include "promise_polyfill.js"
 #endif
 
+// See https://caniuse.com/mdn-javascript_builtins_object_assign
+#if MIN_CHROME_VERSION < 45 || MIN_EDGE_VERSION < 12 || MIN_FIREFOX_VERSION < 34 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED || MIN_SAFARI_VERSION < 90000
+function objAssign(target, source) {
+  for (key in source) {
+    if (source.hasOwnProperty(key)) {
+      target[key] = source[key];
+    }
+  }
+  return target;
+}
+#else
+var objAssign = Object.assign;
+#endif
```

Comment:
Another option here would be to add `{{{ makeObjectAssign() }}}` that would emit `Object.assign` when possible, and a call to `objAssign` otherwise. That is slightly less readable in `src/` but the output code would have `Object.assign` in a more natural way.

Whether that matters, I'm not sure. Especially with closure I'm not sure which way would end up actually shorter. So I'd lean towards simpler code in `src/`, which is probably what you already have here.